scripts/benchmark/checks/deploy_dos.py

```diff
@@ -0,0 +1,296 @@
+import logging
+import pathlib
+from typing import (
+    Tuple
+)
+
+from web3 import (
+    Web3
+)
+
+from eth_utils import (
+    encode_hex,
+    decode_hex,
+)
+
+from eth.constants import (
+    CREATE_CONTRACT_ADDRESS
+)
+from eth.chains.base import (
+    MiningChain,
+)
+from eth.rlp.blocks import (
+    BaseBlock,
+)
+
+from .base_benchmark import (
+    BaseBenchmark,
+)
+from utils.chain_plumbing import (
+    FUNDED_ADDRESS,
+    FUNDED_ADDRESS_PRIVATE_KEY,
+    get_all_chains,
+)
+from utils.compile import (
+    get_compiled_contract
+)
+from utils.reporting import (
+    DefaultStat,
+)
+from utils.tx import (
+    new_transaction,
+)
+
+from abc import (
+    abstractmethod
+)
+
+FIRST_TX_GAS_LIMIT = 367724
+SECOND_TX_GAS_LIMIT = 62050
+THIRD_TX_GAS_LIMIT = 104789
+FORTH_TX_GAS_LIMIT = 21272
+FIFTH_TX_GAS_LIMIT = 21272
+
+CONTRACT_FILE = 'scripts/benchmark/contract_data/DOSContract.sol'
+CONTRACT_NAME = 'DOSContract'
+
+W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
+
+
+class BaseDOSContractBenchmark(BaseBenchmark):
+
+    def __init__(self, num_blocks: int = 20, num_tx: int = 10) -> None:
+        super().__init__()
+
+        self.num_blocks = num_blocks
+        self.num_tx = num_tx
+
+        self.contract_interface = get_compiled_contract(
+            pathlib.Path(CONTRACT_FILE),
+            CONTRACT_NAME
+        )
+
+        self.w3 = Web3()
+
+    def _setup_benchmark(self, chain: MiningChain) -> None:
+        """
+        This hook can be overwritten to perform preparations on the chain
+        that do not count into the measured benchmark time
+        """
+        pass
+
+    @abstractmethod
+    def _apply_transaction(self, chain: MiningChain) -> None:
+        raise NotImplementedError(
+            "Must be implemented by subclasses"
+        )
+
+    def execute(self) -> DefaultStat:
+        total_stat = DefaultStat()
+
+        for chain in get_all_chains():
+
+            self._setup_benchmark(chain)
+
+            value = self.as_timed_result(
+                lambda: self.mine_blocks(chain, self.num_blocks, self.num_tx)
+            )
+
+            total_gas_used, total_num_tx = value.wrapped_value
+
+            stat = DefaultStat(
+                caption=chain.get_vm().fork,
+                total_blocks=self.num_blocks,
+                total_tx=total_num_tx,
+                total_seconds=value.duration,
+                total_gas=total_gas_used,
+            )
+            total_stat = total_stat.cumulate(stat)
+            self.print_stat_line(stat)
+
+        return total_stat
+
+    def mine_blocks(self, chain: MiningChain, num_blocks: int, num_tx: int) -> Tuple[int, int]:
+        total_gas_used = 0
+        total_num_tx = 0
+
+        for i in range(1, num_blocks + 1):
+            block = self.mine_block(chain, i, num_tx)
+            total_gas_used = total_gas_used + block.header.gas_used
+            total_num_tx = total_num_tx + len(block.transactions)
```

Comment:
*mutability nitpick*

Here is how you can do this without all of the incremental mutation.

```python
blocks = tuple(self.mine_block(chain, i, num_tx) for i in range(1, num_blocks + 1))
total_gas_used = sum(block.header.gas_used for block in blocks)
total_num_tx = sum(len(block.transactions) for block in blocks)
```

I find this style easier to read and reason about.

Reply:
fixed :)